src/Playwright/BrowserContext.cs

```diff
@@ -38,6 +64,20 @@ internal BrowserContext(IChannelOwner parent, string guid, BrowserContextInitial
 
         public event EventHandler<IPage> Page;
 
+        public event EventHandler<IRequest> Request;
+
+        public event EventHandler<IRequest> RequestFailed;
+
+        public event EventHandler<IRequest> RequestFinished;
+
+        public event EventHandler<IResponse> Response;
+
+        public ITracing Tracing
+        {
+            get => _tracing;
+            set => throw new NotSupportedException();
+        }
```

Comment:
```suggestion
        public ITracing Tracing {  get; }
```

Reply:
I am following the Page._mouse pattern here.